### PR TITLE
Feature camerafilter base

### PR DESCRIPTION
**Pull request by Nibor9198**
Added and integrated the CameraFilter module.

## Pylint

snake case could not be imported

relative import will be resloved by other pull request

cv2 missing, just a pylint bug

```
cameraFilter.py:1:0: C0103: Module name "cameraFilter" doesn't conform to snake_case naming style (invalid-name)
cameraFilter.py:6:0: E0402: Attempted relative import beyond top-level package (relative-beyond-top-level)
cameraFilter.py:7:0: E0402: Attempted relative import beyond top-level package (relative-beyond-top-level)
cameraFilter.py:8:0: E0402: Attempted relative import beyond top-level package (relative-beyond-top-level)
cameraFilter.py:61:14: E1101: Module 'cv2' has no 'VideoCapture' member (no-member)
cameraFilter.py:66:30: E1101: Module 'cv2' has no 'CAP_PROP_FRAME_WIDTH' member (no-member)
cameraFilter.py:67:30: E1101: Module 'cv2' has no 'CAP_PROP_FRAME_HEIGHT' member (no-member)
cameraFilter.py:82:25: E1101: Module 'cv2' has no 'getRotationMatrix2D' member (no-member)
cameraFilter.py:85:32: E1101: Module 'cv2' has no 'warpAffine' member (no-member)
cameraFilter.py:88:29: E1101: Module 'cv2' has no 'getRectSubPix' member (no-member)
cameraFilter.py:95:30: E1101: Module 'cv2' has no 'resize' member (no-member)
cameraFilter.py:99:20: E1101: Module 'cv2' has no 'waitKey' member (no-member)
cameraFilter.py:107:8: E1101: Module 'cv2' has no 'destroyAllWindows' member (no-member)```

### CODE REVIEW

File: vgc/cameraFilter.py [hunk 1]
```diff
@@ -1,82 +1,107 @@
 """Filters a video stream.
```
Comment (emilcw) [inline vgc/cameraFilter.py:1]: This should list all classes, exceptions, functions and objects that belong to the module. Add this. See "Dokumentering av kod" for more info.

File: tests/test_cameraFilter.py [hunk 2]
```diff
@@ -1,8 +1,15 @@
-
-from vgc.cameraFilter import cameraFilter 
+from time import sleep
+from vgc.camera_filter import CameraFilter 
 
 def test_camerafilter():
-    cf = cameraFilter([])
+    """Testing the Came"""
```
Comment (emilcw) [inline tests/test_cameraFilter.py:5]: Try to explain more clearly what this fucntion actually tests.

File: vgc/cameraFilter.py [hunk 3]
```diff
@@ -1,82 +1,107 @@
...
+class CameraFilter(Filter):
     """Filters a video stream
-            
-        This filter rotates and cropps a video stream according to a pitch, a jaw and a zoom variable . 
-        The pitch, jaw and zoom variables are updated asynchronously through the update funciton.
-    
+
+        This filter rotates and cropps a video stream according to
+        a pitch, a jaw and a zoom variable. The pistch, jaw and zoom
+        variables are updated asynchronously through the update funciton.
```
Comment (emilcw) [inline vgc/cameraFilter.py:16]: This docstring should list functions and variables that belong to the class, add that. See "Dokumentering av kod" for more info. Would be nice with further explanation to the variables.

File: vgc/cameraFilter.py [hunk 4]
```diff
@@ -1,82 +1,107 @@
...
+                rotated_frame = cv2.warpAffine(frame, matrix,
+                                               (w_frame, h_frame))
                 # Crop the frame
-                crop_frame = cv2.getRectSubPix(rotated_frame, (int(w*self.zoom_in), int(h*self.zoom_in)), (w_frame/2, h_frame/2 - h*self.zoom_in/2 + self.pitch_in))
+                crop_frame = cv2.getRectSubPix(rotated_frame,
+                                                (int(width*self.zoom_in),
+                                                 int(height*self.zoom_in)),
+                                                (w_frame/2, h_frame/2
+                                                 - height*self.zoom_in/2
+                                                 + self.pitch_in))
```
Comment (emilcw) [inline vgc/cameraFilter.py:93]: These lines are too long, max 79 char.

File: tests/test_cameraFilter.py [hunk 5]
```diff
@@ -1,8 +1,15 @@
-
-from vgc.cameraFilter import cameraFilter 
+from time import sleep
```
Comment (emilcw) [inline tests/test_cameraFilter.py:1]: The test file should also follow the PEP-8 standard. Make sure you add docstrings etc

**Review by emilcw — CHANGES_REQUESTED** [review]
Great work! Some minor fixes is needed tho.

File: tests/test_cameraFilter.py (hunk 5 above)
Comment (aannalarsson) [reply tests/test_cameraFilter.py:1]: The tests for the camera filter are not done and therefore not pushed.

File: tests/test_cameraFilter.py (hunk 2 above)
Comment (aannalarsson) [reply tests/test_cameraFilter.py:5]: The tests for the camera filter are not done and therefore not pushed.

File: vgc/cameraFilter.py (hunk 1 above)
Comment (aannalarsson) [reply vgc/cameraFilter.py:1]: We have added this to the docstring.

File: vgc/cameraFilter.py (hunk 4 above)
Comment (aannalarsson) [reply vgc/cameraFilter.py:93]: Fixed!

Comment (Nibor9198) [reply vgc/cameraFilter.py:93]: None of our lines are longer than 74

File: tests/test_cameraFilter.py (hunk 5 above)
Comment (Nibor9198) [reply tests/test_cameraFilter.py:1]: The test file shouldn't have been part of this pullrequest at all

**Review by emilcw — APPROVED** [review]
Approved by Emil and Carl. @DennisBerntsson

**Pull request merged**